stacks-signer/src/stacks_client.rs

```diff
@@ -98,8 +142,157 @@ impl StacksClient {
         // See: https://github.com/stacks-network/stacks-blockchain/issues/3921
         SLOTS_PER_USER
     }
+
+    fn seralize_sign_tx_anchor_mode_version(
```

Comment:
It might make more sense to have one function for non-sponsored transactions, and one for sponsored transactions.  Then, the caller won't have to supply `None` gratuitously.